### DISCUSSION TITLE
Why is audio being stripped off?

### DISCUSSION THREAD

**Discussion by ryanjkendrick** [Zulko/moviepy#1631] [Q&A]
Is it obvious to anyone why the code listed below strips off audio on the exported file?

https://github.com/rhinohq/GoProAssistant/blob/main/TextVid/TextVid.py

**Comment by Zulko** [comment]
Yes, it is because `set_audio` is not in-place, `clip.set_audio()` doesnt modify `clip`, but rather returns another clip. Your last line should be:

```
result = CompositeVideoClip(video_clips).set_audio(video.audio)
```

Very sorry for the misleading naming, as I understand, this will be fixed in v2.0.

**Reply by ryanjkendrick** [reply]
Thanks very much for your speedy reply. 

I am afraid I have tried your suggestion and it has not worked. The original file definitely has audio but the outputted file does not. I have committed to the aforementioned script to show my changes.

**Comment by keikoro** [comment]
@rhinohq I'm confused about why this was posted here. Notifications for Discussions don't show up in the same place as those for issues and I think it would have been better to create an issue for this as this sounds like a regular issue... which we could have labelled accordingly.

**Comment by keikoro** [comment]
There also doesn't appear to be a way to actually transfer this to the issue tracker, so I'm afraid this will continue to sit here largely unnoticed.
